### ISSUE TITLE
Method 'BanUserAsync' in type 'Discord.WebSocket.SocketGuild' from assembly 'Discord.Net.WebSocket`

### ISSUE THREAD

**Issue by binaryn3xus** [zobweyt/Discord.NET-Template#3]
Opened the project, Added my token and guild id to the appsettings.json and to the usersecrets. Still get this error every time.

```
Unhandled exception. System.TypeLoadException: Method 'BanUserAsync' in type 'Discord.WebSocket.SocketGuild' from assembly 'Discord.Net.WebSocket, Version=3.13.0.0, Culture=neutral, PublicKeyToken=null' does not have an implementation.
   at Discord.WebSocket.DiscordSocketClient..ctor(DiscordSocketConfig config, DiscordSocketApiClient client, DiscordShardedClient shardedClient, DiscordSocketClient parentClient)
   at Discord.WebSocket.DiscordSocketClient..ctor(DiscordSocketConfig config)
   at Discord.Addons.Hosting.Injectables.InjectableDiscordSocketClient..ctor(IOptions`1 config)
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeDirectByRefWithFewArgs(Object obj, Span`1 copyOfArgs, BindingFlags invokeAttr)
   at System.Reflection.MethodBaseInvoker.InvokeWithOneArg(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitIEnumerable(IEnumerableCallSite enumerableCallSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(ServiceCallSite callSite, ServiceProviderEngineScope scope)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.CreateServiceAccessor(ServiceIdentifier serviceIdentifier)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(ServiceIdentifier serviceIdentifier, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
   at Microsoft.Extensions.Hosting.Internal.Host.StartAsync(CancellationToken cancellationToken)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(IHost host, CancellationToken token)
   at Program.<Main>$(String[] args) in E:\Git\Discord.Net.Template\src\Template\Program.cs:line 50
   at Program.<Main>(String[] args)
dotnet watch ❌ Exited with error code -532462766
```

**Comment by zobweyt** [comment]
Hey! Thanks for using the template!

The issue you are encountering is associated with the internal changes in Discord.NET 3.15.0. Please see the **important notes** section [here](https://github.com/discord-net/Discord.Net/releases/tag/3.15.0) for detailed information. In addition, since the template is built using the [Discord.Addons.Hosting](https://github.com/Hawxy/Discord.Addons.Hosting), these changes are required on their side.

**TLDR: Revert to Discord.NET version 3.14.0. Alternatively, you can also copy or [use](https://github.com/zobweyt/Discord.Net.Template/generate) the template again, as the necessary fixes have been committed.**

Let me know, if it fixes this issue.

**Comment by binaryn3xus** [comment]
Thank you so much. That fixed the problem perfectly. And thank you so much for keeping this Template up to date, it is a great resources for beginners getting into Discord Bots.
